.github/workflows/publish.yaml

```
@@ -29,6 +38,7 @@ jobs:
 
   publish-to-testpypi:
     name: Publish distribution package to TestPyPI
+    if: github.ref == 'refs/heads/master'
```

I like using the `github.ref_name` var (it was added later than I first wrote the guide, I think: `github.ref_name` and `github.ref_type` are a bit more convenient). Also, there's a way to get the repo default branch w/o having to hardcode it, been using it for years.
```suggestion
    if: >-
      github.ref_name == github.event.repository.default_branch
```